File: java/src/test/java/mx/iteso/YogurtIceCreamStoreTest.java
package mx.iteso;

import static org.junit.Assert.assertEquals;
import org.junit.Test;
import mx.iteso.Cone;
import mx.iteso.stores.YogurtIceCreamStore;

public class YogurtIceCreamStoreTest {
    @Test
    public void yogurtIceCreamStoreGuavaTest() {
        Cone cone = new YogurtIceCreamStore().prepareCone("Guava");
        assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Guayaba");
    }

    @Test
    public void yogurtIceCreamStoreMangoTest() {
        Cone cone = new YogurtIceCreamStore().prepareCone("Mango");
        assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Mango");
    }

    @Test
    public void yogurtIceCreamStorePineappleTest() {
        Cone cone = new YogurtIceCreamStore().prepareCone("Pineapple");
        assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Piña");
    }

    @Test
    public void yogurtIceCreamStoreStrawberryTest() {
        Cone cone = new YogurtIceCreamStore().prepareCone("Strawberry");
        assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Fresa");
    }
}

File: python/src/unittest/python/YogurtIceCreamStoreTest.py
from mockito import mock, verify, when
import unittest
from store.YogurtIceCreamStore import *


class YogurtIceCreamStoreTest(unittest.TestCase):

    def setUp(self):
        self.store = YogurtIceCreamStore()

    def guavaYogurtIceCreamTest(self):
        cone = self.store.prepareCone("Guava")
        self.assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Guayaba")

    def mangoYogurtIceCreamTest(self):
        cone = self.store.prepareCone("Mango")
        self.assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Mango")

    def pineappleYogurtIceCreamTest(self):
        cone = self.store.prepareCone("Pineapple")
        self.assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Piña")

    def strawberryYogurtIceCreamTest(self):
        cone = self.store.prepareCone("Strawberry")
        self.assertEquals(cone.getIceCream().getFlavor(), "Yogurt -> Fresa")